server/src/main/java/io/crate/analyze/GenericPropertiesConverter.java

```diff
@@ -89,6 +89,18 @@ static void settingsFromProperties(Settings.Builder builder,
                                        boolean setDefaults,
                                        Predicate<String> ignoreProperty,
                                        String invalidMessage) {
+        for (Map.Entry<String, Object> property: properties.properties().entrySet()) {
+            if (property.getValue() == null) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        Locale.ENGLISH,
+                        "Cannot set NULL to property %s. Use ALTER TABLE RESET to reset to default value.",
+                        property.getKey()
+                    )
+                );
+            }
+        }
+
```

Comment:
Could do the catch further below:

```diff
diff --git a/server/src/main/java/io/crate/analyze/GenericPropertiesConverter.java b/server/src/main/java/io/crate/analyze/GenericPropertiesConverter.java
index 5454354dc4f..a1d68ad675d 100644
--- a/server/src/main/java/io/crate/analyze/GenericPropertiesConverter.java
+++ b/server/src/main/java/io/crate/analyze/GenericPropertiesConverter.java
@@ -103,11 +103,15 @@ static void settingsFromProperties(Settings.Builder builder,
             }
             SettingHolder settingHolder = getSupportedSetting(supportedSettings, settingName);
             if (settingHolder == null) {
                 throw new IllegalArgumentException(String.format(Locale.ENGLISH, invalidMessage, entry.getKey()));
             }
-            settingHolder.apply(builder, entry.getValue());
+            Object value = entry.getValue();
+            if (value == null) {
+                ..
+            }
+            settingHolder.apply(builder, value);
         }
```

To avoid iterating twice.

I'm not sure about including the `ALTER TABLE` part in the error message if we do the validation here like this. `GenericPropertiesConverter` should be generally usable for any `GenericProperties`, not just in the context of `ALTER TABLE`. The message could be misleading if used in another context.
We either have to raise the error elsewhere in a alter-table context, parameterize it or make it more general.